analyses/fusion_filtering/02-fusion-filtering.R

```diff
@@ -0,0 +1,269 @@
+#
+
+  
+				
+				
+				
+
+library("optparse")
+library("dplyr")
+library("tidyr")
+library("purrr")
+library("reshape2")
+
+option_list <- list(
+  make_option(c("-S", "--standardFusionFile"),type="character",
+              help="Standardized fusion calls from (.TSV) "),
+  make_option(c("-e","--expressionMatrix"),type="character",
+              help="Matrix of expression for samples in standardFusion file (.RDS) "),
+  make_option(c("-r", "--reathroughFilter"), type="character",action = "store_true",
+              help="Boolean to filter readthrough",default=TRUE),
+  make_option(c("-t","--expressionFilter"), type="integer",
+              help="threshold for TPM/FPKM filter",default=1),
+  make_option(c("-a","--artifactFilter"),type="character",
+              help="red flag filter from Annotation ; in OpenPBTA annotation is from FusionAnnotator"),
+  make_option(c("-j","--junctionReadCountFilter"), type="integer",
+              help="threshold for JunctionReadCount"),
+  make_option(c("-s","--spanningFragCountFilter"),type="integer",
+              help="threshold for (SpanningFragCount - JunctionReadCount)"),
+  make_option(c("-i","--readingFrameFilter"),type="character",
+               help="reading frame filtering ( regex to capture inframe|frameshift|other)"),
+  make_option(c("-R","--referenceFolder"),type="character",
+                help="reference folder with required gene lists"),
+  make_option(c("-o","--outputfile"),type="character",
+              help="Filtered fusion calls (prefix for _QC_expression_filtered_annotated.RDS)")
+)
+
+
+# get command line options, if help option encountered print help and exit,
+
+opt <- parse_args(OptionParser(option_list=option_list))
+
+standardFusionFile<-opt$standardFusionFile
+expressionMatrix<-opt$expressionMatrix
+reathroughFilter<-opt$reathroughFilter
+expressionFilter<-opt$expressionFilter
+artifactFilter<-opt$artifactFilter
+junctionReadCountFilter<-opt$junctionReadCountFilter
+spanningFragCountFilter<-opt$spanningFragCountFilter
+referenceFolder<-opt$referenceFolder
+readingFrameFilter<-opt$readingFrameFilter
+
+# read standardized fusion calls
+standardFusioncalls<-read.delim(standardFusionFile,stringsAsFactors=FALSE)
+# to obtain geneA and geneB for gene search below
+standardFusioncalls <- cbind(standardFusioncalls, colsplit(standardFusioncalls$FusionName, pattern = '--', names = c("GeneA","GeneB")))
+# Intergenic fusion will have Gene1A,Gene2A,Gene1B,Gene2B
+standardFusioncalls <- cbind(standardFusioncalls, colsplit(standardFusioncalls$GeneA, pattern = '/', names = c("Gene1A","Gene2A")))
+standardFusioncalls <- cbind(standardFusioncalls, colsplit(standardFusioncalls$GeneB, pattern = '/', names = c("Gene1B","Gene2B")))
+
+#formatting dataframe for filtering
+standardFusioncalls<-standardFusioncalls %>% 
+  #For annotation later
+  mutate (Gene1A_anno=NA,Gene2A_anno=NA,Gene1B_anno=NA,Gene2B_anno=NA) %>% 
```

Comment:
I don't think you need this line because you will be dropping these as part of the expression filtering and adding them back in when doing the annotation: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/115/files#diff-b69f57dea3b73984f562cedb95b336d4R238-R243

Reply:
you are right, I can edit this! This is also why the is.na() as the condition was incorrect earlier.